osquery/main/shell.cpp

```diff
@@ -83,6 +88,42 @@ int profile(int argc, char *argv[]) {
   return 0;
 }
 
+// readline completion expects strings to be malloced. readline will free them
+// later.
+char *copy_string(const std::string &str) {
+  char *copy = NULL;
+  if ((copy = (char *)malloc(str.size() + 1)) == NULL) {
+    LOG(ERROR) << "Malloc failed. Exiting!";
```

Comment:
Let's make this a bit more helpful. "Cannot allocate memory for shell auto-completion"